src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T [] array = (T[]) new  Object [10];
```

Comment:
The initial capacity of the array (10) is a magic number. Please define it as a constant to improve code readability.